numba/core/boxing.py

```diff
@@ -265,6 +265,12 @@ def box_bytes(typ, val, c):
     return c.pyapi.bytes_from_string_and_size(obj.data, obj.nitems)
 
 
+@box(types.ByteArray)
```

Comment:
Would guess that this and `@box(types.Bytes)` need to look something like:
```python
@box(<bytes-like-type>)
def box_bytes(typ, val, c):
    obj = c.context.make_helper(c.builder, typ, val)
    ret = c.pyapi.<bytes-like ctor call>(obj.data, obj.nitems)
    c.context.nrt.decref(c.builder, typ, val)
    return ret
```

Reply:
This was 100% the issue. Thank you so much.